sql/core/src/main/scala/org/apache/spark/sql/execution/command/tables.scala

```diff
@@ -1342,7 +1342,7 @@ case class ShowCreateTableAsSerdeCommand(
     storage.serde.foreach { serde =>
       builder ++= s"ROW FORMAT SERDE '$serde'\n"
 
-      val serdeProps = conf.redactOptions(metadata.storage.properties).map {
+      val serdeProps = conf.redactOptions(metadata.storage.properties).toSeq.sortBy(_._1).map {
```

Comment:
https://github.com/apache/spark/blob/1cb8250d166f5d877f24f6ea097d27d7168ecf15/sql/core/src/main/scala/org/apache/spark/sql/execution/command/tables.scala#L1202-L1210

`showDataSourceTableOptions` already do the similar work in https://github.com/apache/spark/pull/34753